ListUtil.xs

```diff
@@ -623,44 +623,26 @@ PPCODE:
 {
     int size = 0;
     int start = 0;
-    int end = 0;
-    int i = 0;
 
-    size = SvIV( ST(0) );
+    if (items > 1) {
+        size = SvIV(*(SP + 1));
 
-    if ( ix == 0 ) {
-        start = 1;
-        end = start + size;
-        if ( size < 0 ) {
-            end += items - 1;
-        }
-        if ( end > items ) {
-            end = items;
-        }
-    }
-    else {
-        end = items;
-        if ( size < 0 ) {
-            start = -size + 1;
-        }
-        else {
-            start = end - size;
-        }
-        if ( start < 1 ) {
-            start = 1;
+        if (ix) start = items - size - 1;
+        start = 1 + (start + items - 1) % (items - 1);
+
+        if (size > items - 1) {
+            size = items - 1;
+        } else if (size < 0) {
+            size = size + items - 1;
         }
     }
 
-    if ( end < start ) {
-        XSRETURN(0);
-    }
-    else {
-        EXTEND( SP, end - start );
-        for ( i = start; i <= end; i++ ) {
-            PUSHs( sv_2mortal( newSVsv( ST(i) ) ) );
-        }
-        XSRETURN( end - start );
+    while (--size >= 0) {
+        ++SP;
+        *SP = *(SP + start);
```

Comment:
Hmm.. This is quite the change of behaviour - we're now returning aliases of the original input list, rather than new temporary SVs that copy it. This would have a change of behaviour in lvalue context; for example

    my @n = (1 .. 10);
    $_++ for tail 4, @n;

Reply:
If such code wouldn't work before, I'd say this would just be a new feature...

Reply:
It could be considered a feature yes. But it's a somewhat surprising change of behaviour. Is there a chance of accidental collateral damage?

Reply:
Returning aliases is the behavior I would expect.